backend/src/nodes/nodes/utility/random_number.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+from random import seed, randint
+from typing import Union
+
+from . import category as UtilityCategory
+
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import NumberInput
+from ...properties.outputs import NumberOutput
+
+
+@NodeFactory.register("chainner:utility:random_number")
+class RandomNumberNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Generate a random integer."
+        self.inputs = [
+            NumberInput(
+                "Minimum Value",
+                minimum=None,
+                maximum=None,
+                precision=100,
+                controls_step=1,
+            ),
+            NumberInput(
+                "Maximum Value",
+                minimum=None,
+                maximum=None,
+                precision=100,
+                controls_step=1,
+            ),
+            NumberInput(
+                "Seed",
+                minimum=-1,
+                maximum=None,
+                default=-1,
+                precision=100,
+                controls_step=1,
+            )
+        ]
+        self.outputs = [
+            NumberOutput("Result", output_type="number")

```

Comment:
We actually know quite a bit about the output type. It's not just any number, we know that it's an integer within the range given by the above inputs.

Don't worry about the below expression. It's a [Navi type expression](https://github.com/chaiNNer-org/Navi/blob/main/navi.md), but you can also treat it as a black box for now :)
```suggestion
            NumberOutput("Result", output_type="int & max(.., Input0) & min(.., Input1)")

```